src/modules/Main.ui.js

```diff
@@ -215,27 +215,24 @@ export default class Main extends Component<Props, State> {
         delay: 500
       })
     })
-    if (Platform.OS === 'android') {
-      Linking.getInitialURL().then(url => {
-        if (url) {
-          this.props.urlRecived(url)
-        }
-        // this.navigate(url);
-      })
-    } else {
-      Linking.addEventListener('url', this.handleOpenURL)
-    }
+    Linking.getInitialURL().then(url => {
+      if (url) {
+        this.doDeepLink(url)
+      }
+      // this.navigate(url);
+    }).catch(err => console.log('error occurred ', err))
+    Linking.addEventListener('url', this.handleOpenURL)
+  }
+  doDeepLink (url: string) {
+    const parsedUri = URI.parse(url)
+    const query = parsedUri.query
+    const splitArray = query.split('token=')
```

Comment:
This is missing the point, Allen. `uri-js` will parse all the elements of the URI for you, you are still making the bad assumption that everything after `token=` is the token. If this URI had more parameters this would break. Please use uri-js for all parsing.

Reply:
Paul, I disagree, 
When I parse this url edge://recovery?token=IamToken&nextVar=nextVar
I get the following object 
```
{
  fragment: undefined,
  host: “recovery”
  path: “”,
  port: undefined,
  query: “token=IamToken&nextVar=nextVar”
  reference: “absolute
  scheme:”edge”
  userinfo: undefined
}
```

My code then takes the query and splits it. 
I then take into consideration any additional items in the query string and trim them. The one thing I did not consider was if ‘token=‘ was even present in the query string, so I am now checking for that.
I have also added a const called token for the value that is finalArray[0]
So running the following code here are my tests and results. 

```
const parsedUri = URI.parse(url)
const query = parsedUri.query
if (!query.includes('token=')) {
 return
}
const splitArray = query.split('token=')
const nextString = splitArray[1]
const finalArray = nextString.split('&')
const token = finalArray[0]
console.log(token)
```

Test url: edge://recovery?token=IamToken&nextVar=nextVar
Result of console log: IamToken

Test url: https://recovery.edgesecure.co/recovery?token=IamToken&nextVar=nextVar
Result of console log: IamToken

Test url: https://recovery.edgesecure.co/recovery?bob=bob&token=IamToken&nextVar=nextVar
Result of console log: IamToken

Test url: https://recovery.edgesecure.co/recovery?bob=bob&nextVar=nextVar
Result of console log: nothing, the function returns before we get the console log 

Based on my results I have covered the all the different use cases. I did not see anything readily apparent in the uri-js library that allows me to just pluck a value out of the query string. If I had seen that I would have done it.